### PR TITLE
dbex/76057 radio button conversion

### LINKED ISSUE #76057: Update radio buttons to use web components
## Issue Description
- Timeboxed story since there are so many radio buttons on 526. 
- Do a first pass to try and upgrade some radio buttons from widgets to web components. yesNoPattern may be a good place to start.
- Make sure to use prop `uswds=false`. We will enable v3 on all radio buttons at a later time for UI/UX consistency.
- Verify unit tests and e2e tests are still successful. Update as needed.

**Scope** of this ticket is to complete one set of radio buttons on one page.

From Platform Design Team - https://dsva.slack.com/archives/C03R5SBELQM/p1708966195201799
We made the decision last week to delay the V3 opt out due to vets-website CI/CD stability issues. Now that things have normalized a little more, we will be proceeding with our plan to make V3 web components opt out only. The switch will occur on March 4th.
On March 4th, all web components in vets-website and content build will render their V3 variants.
Additionally, the design system team originally targeted 4/19 as the date to officially retire and remove V1 components completely. In light of the impact recent CI/CD issues introduced for VFS teams, the retirement and removal date for v1 components has been moved to **May 3 (5/3).**

## Front end tasks
- Create a list of all pages with radio buttons that need to be upgraded
- Ensure the page is styled according to design spec. 
- ~Ensure the back end data from API is displayed in local and review instance~
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin
- Test for color contrast and color blindness issues
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only

## Acceptance Criteria
- [ ] Create a ticket for the rest of the radio buttons upgrade for team 2 
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] The data returned matches API response (for given user or scenario)
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by michael-wojcik**
- **76057 convert radio buttons on terminallyIll page**
- **👌 IMPROVE: switch from Enzyme to RTL for testing terminallyIll component**
- **👌 IMPROVE: convert alternateNames from React to web component**
- **👌 IMPROVE: convert trainingPay from React to web component**
- **👌 IMPROVE: add terminallyIll to maxmimal-test in E2E test**
- **👌 IMPROVE: convert evidenceTypes YesNo radio from React to web component**
- **👌 IMPROVE: update unit tests for trainingPay**
- **🐛 FIX: update terminallyIll unit tests**
- **👌 IMPROVE: update unit tests for evidenceTypes**

## Summary

- Updated a few of the Yes/No radio button components on 526EZ to use the new web component widget.
- This PR will serve as an example to future tickets related to similar UI updates.
- DBEX team
- No flipper being used

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#76057

## Testing done

Modernized unit tests corresponding to the updated widgets:
- now using React Testing Library, in line with Platform's updated standards;
- now using more user-centric selectors/behaviors to simulate UX, as opposed to pure process testing

Added one E2E test as well, adding `terminallyIll` to the `maximal` E2E test.

Additional manual smoke testing done to confirm same look and feel after updating components.

To verify changes, unit tests should be passing and new `<va-radio>` element should be appearing on the associated pages.

## Screenshots

| Before | After |
| ------ | ----- |
| alternateName--old | alternateName--new |
| terminallyIll--old | terminallyIll--new |
| trainingPay--old | trainingPay--new |
| evidenceTypes--old | evidenceTypes--new |

## What areas of the site does it impact?

526EZ form

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by powellkerry — DISMISSED** [review]

File: src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx [hunk 1]
```diff
@@ -22,75 +27,82 @@ describe('evidenceTypes', () => {
...
-    form.find('form').simulate('submit');
+    $('va-radio', container).__events.vaValueChange({
+      detail: { value: 'Y' },
+    });
+    const submitButton = getByText(/submit/i);
+    userEvent.click(submitButton);
+    const error = getByText(
+      /please select at least one type of supporting evidence/i,
+    );
+    expect(error).to.exist;
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx:76]: The `getByText` query throws an exception if there's no match. It should be sufficient to just call `getByText`.

File: src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx [hunk 2]
```diff
@@ -22,75 +27,82 @@ describe('evidenceTypes', () => {
...
+    userEvent.click(submitButton);
     expect(onSubmit.calledOnce).to.be.true;
-    expect(form.find('.usa-input-error-message').length).to.equal(0);
-    form.unmount();
+    try {
+      getByText(/please select at least one type of supporting evidence/i);
+    } catch (e) {
+      // Error text in try block should not be present.
+      expect(e).to.exist;
+    }
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/pages/evidenceTypes.unit.spec.jsx:106]: in the previous test this was used `expect($('va-radio').error).to.be.null`. could we use that again?

File: src/applications/disability-benefits/all-claims/tests/pages/terminallyIll.unit.spec.jsx [hunk 3]
```diff
@@ -23,16 +25,20 @@ describe('Terminally Ill', () => {
...
-    form.unmount();
+    getByText('High Priority claims');
+
+    const question = container.querySelector('va-radio');
+    expect(question).to.have.attribute('label', 'Are you terminally ill?');
+
+    expect(container.querySelector('va-radio-option[label="Yes"', container)).to
+      .exist;
+    expect(container.querySelector('va-radio-option[label="No"', container)).to
+      .exist;
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/pages/terminallyIll.unit.spec.jsx:36]: Really like how on this "it should render" test we're checking for the presence of 1. the question (label) and 2. each radio button and their respective label. Is this something we could do for each of these pages?

File: src/applications/disability-benefits/all-claims/tests/pages/trainingPay.unit.spec.jsx [hunk 4]
```diff
@@ -42,30 +46,32 @@ describe('Training Pay', () => {
...
     );
 
-    form.find('form').simulate('submit');
-    expect(form.find('.usa-input-error-message').length).to.equal(1);
+    const submitButton = getByText(/submit/i);
+    userEvent.click(submitButton);
+    // Check that form is not submitted due to missing answer.
     expect(onSubmit.called).to.be.false;
-    form.unmount();
+    expect($('va-radio').error).to.exist;
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/pages/trainingPay.unit.spec.jsx:53]: Do we want to look for specific error message?

**Comment by christinec-fftc** [comment]
hmm, I was manual testing on my local and it looks like the `uswds: false` prop is not being honored. However it's debatable whether that's worth fixing and using at this point.

i'm also seeing this lovely border. this we do need to fix prior to merging
image

i'm going to put this pr in 'draft' status just so it does not accidentally get merged while we're still working on it.

**Review by christinec-fftc — APPROVED** [review]

**Pull request merged**